handler.go

```diff
@@ -342,3 +310,51 @@ func closeSendStream(ctx context.Context, stream *yarpctransport.ClientStream) e
 	}
 	return nil
 }
+
+// streamIOHandler uses provided stream message reader to provide stream
+// IO methods to read requests one by one and handle responses.
+type streamIOHandler struct {
+	out             output
+	serializer      encoding.Serializer
+	streamMsgReader encoding.StreamRequestReader
+}
+
+// NextRequest returns the next stream request body from the given stream
+// message reader.
+// Note: this method must not be called once EOF has been returned.
+func (s *streamIOHandler) NextRequest() ([]byte, error) {
+	msg, err := s.streamMsgReader.NextBody()
+	if err == io.EOF {
+		return nil, err
+	}
+	if err != nil {
+		return nil, fmt.Errorf("Failed while reading stream input: %v", err)
+	}
+
+	return msg, nil
+}
+
+// HandleResponse validates the response bytes and prints indented JSON body.
```

Comment:
Is there any reason you have used the word `validates`? What does it mean in this context? From what I see, we get the response, we deserialize it and we print it.

Reply:
You are right `validates` isn't the right word, it parses the response bytes and prints it